cmd/lotus-storage-miner/sectors.go

```diff
@@ -403,6 +407,100 @@ var sectorsRefsCmd = &cli.Command{
 	},
 }
 
+var sectorsExtendCmd = &cli.Command{
+	Name:      "extend",
+	Usage:     "Extend sector expiration",
+	ArgsUsage: "<sectorNumbers...>",
+	Flags: []cli.Flag{
+		&cli.Int64Flag{
+			Name:     "new-expiration",
+			Usage:    "new expiration epoch",
+			Required: true,
+		},
+		&cli.StringFlag{},
+	},
+	Action: func(cctx *cli.Context) error {
+		nodeApi, closer, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		api, nCloser, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer nCloser()
+
+		ctx := lcli.ReqContext(cctx)
+		if !cctx.Args().Present() {
+			return xerrors.Errorf("must pass at least one sector number")
+		}
+
+		maddr, err := nodeApi.ActorAddress(ctx)
+		if err != nil {
+			return xerrors.Errorf("getting miner actor address: %w", err)
+		}
+
+		sectors := map[miner.SectorLocation][]uint64{}
+
+		for i, s := range cctx.Args().Slice() {
+			id, err := strconv.ParseUint(s, 10, 64)
+			if err != nil {
+				return xerrors.Errorf("could not parse sector %d: %w", i, err)
+			}
+
+			p, err := api.StateSectorPartition(ctx, maddr, abi.SectorNumber(id), types.EmptyTSK)
+			if err != nil {
+				return xerrors.Errorf("getting sector location for sector %d: %w", id, err)
+			}
+
+			if p == nil {
+				return xerrors.Errorf("sector %d not found in any partition", id)
+			}
+
+			sectors[*p] = append(sectors[*p], id)
+		}
+
+		params := &miner0.ExtendSectorExpirationParams{}
+		for l, numbers := range sectors {
+
+			params.Extensions = append(params.Extensions, miner0.ExpirationExtension{
+				Deadline:      l.Deadline,
+				Partition:     l.Partition,
+				Sectors:       bitfield.NewFromSet(numbers),
+				NewExpiration: abi.ChainEpoch(cctx.Int64("new-expiration")),
```

Comment:
Would be nice to sanity-check this param (positive / greater than all the current expiration epochs / whatever actors code expects)

Reply:
Pretty sure that MpoolPush will fail with a nice error message when that happens